core/src/main/java/tc/oc/pgm/api/PGM.java

```diff
@@ -35,6 +36,8 @@
 
   ScheduledExecutorService getAsyncExecutor();
 
+  VanishManager getVanishManager();
+
```

Comment:
This makes PGM depend on the vanish feature being present, which goes against #326 since it currently is a community feature. Maybe check for the vanish value in the same way you intended for the util package? (See other comment for where i'm thinking about)

Reply:
Hmm, I'd say make `VanishManager` an interface in `tc.oc.pgm.api`. With 2 implementation, a `NoopVanishManager` in `pgm` and a `VanishManagerImpl` in `community`.